src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -682,6 +743,107 @@ function InfoBottomSheet({
   )
 }
 
+function ActionCard({ action }: { action: BeforeDepositAction }) {
+  return (
+    <Touchable
+      style={styles.touchable}
+      key={action.name}
+      borderRadius={20}
+      onPress={action.onPress}
+      testID={`Earn/BeforeDepositBottomSheet/${action.name}`}
+    >
+      <>
+        <action.iconComponent color={Colors.black} />
+        <View style={styles.cardContainer}>
+          <Text style={styles.actionTitle}>{action.title}</Text>
+          <Text style={styles.actionDetails}>{action.details}</Text>
+        </View>
+      </>
+    </Touchable>
+  )
+}
+
+function BeforeDepositBottomSheet({
```

Comment:
nit: consider moving this component to it's own file, this file is getting too big